StripeConnect/StripeConnectTests/Internal/Webview/ConnectWebViewTests.swift

```diff
@@ -0,0 +1,167 @@
+//
+//  ConnectWebViewTests.swift
+//  StripeConnectTests
+//
+//  Created by Chris Mays on 8/14/24.
+//
+
+import Foundation
+
+@_spi(STP) import StripeCore
+@testable import StripeConnect
+import XCTest
+import WebKit
+import SafariServices
+
+class ConnectWebViewTests: XCTestCase {
+    
+    var mockURLOpener: MockURLOpener!
+    var webView: ConnectWebView!
+    
+    override func setUp() {
+        mockURLOpener = .init()
+        webView = ConnectWebView(frame: .zero,
+                                 configuration: .init(),
+                                 urlOpener: mockURLOpener,
+                                 bundleVersion: "1.2.3")
+    }
+    
+    func testUserAgent() {
+        // Create an expectation for the asynchronous operation
+        let expectation = XCTestExpectation(description: "User Agent Fetched")
+        webView.evaluateJavaScript("navigator.userAgent") { (result, error) in
+            if let error = error {
+                XCTFail("Error fetching user agent: \(error.localizedDescription)")
+                return
+            }
+            
+            guard let userAgent = result as? String else {
+                XCTFail("User agent is not a string")
+                return
+            }
+                        
+            XCTAssertTrue(userAgent.hasSuffix("- stripe-ios/1.2.3"), "User agent should include the SDK identifier")
+            expectation.fulfill()
```

Comment:
nit: I think we want the expectation to fulfill, regardless of whether there's failures. XCTest continues to run the test, even after a failure occurs. So by returning early without fulfilling the expectation, the test can potentially take the full 2s to run before timing out.

You could do this by including a `defer { expectation.fulfill() }` or you could alternatively simplify the XCT checks:

```suggestion
            XCTAssertNil(error, "Error fetching user agent: \(error.localizedDescription)")
            XCTAssertEqual((result as? String).hasSuffix("- stripe-ios/1.2.3"), true, "User agent should include the SDK identifier but value was: \(String(describing: result))")
            expectation.fulfill()
```

^ I also included the user agent in the failure message. Then we won't have to insert breakpoints while running the test to figure out what the actual value.